_posts/2019-11-01-deploying-debezium-with-kafkaconnector-resource.md

```diff
@@ -0,0 +1,697 @@
+    apply -f https://raw.githubusercontent.com/strimzi/strimzi-kafka-operator/0.15.0/examples/kafka/kafka-persistent-single.yaml \
+  && kubectl wait kafka/my-cluster --for=condition=Ready --timeout=300s -n kafka
+```
+
+What we've got so far looks like this
+
+<pre style="line-height: 1.2 !important;"><code>
+┌────────────────────────────────┐         ┌────────────────────────────────┐
+│ minikube, namespace: kafka     │         │ docker                         │
+│                                │         │                                │
+│ Kafka                          │         │ MySQL                          │
+│  name: my-cluster              │         │                                │
+│                                │         └────────────────────────────────┘
+│ KafkaConnect                   │
+│  name: my-connect              │
+│                                │
+└────────────────────────────────┘
+</code></pre>
+
+> Don't worry, the ASCII art gets a lot better than this!
+
+What's missing from this picture is Kafka Connect in the minikube box.
+
+# Kafka Connect image
+
+The next step is to [create a Strimzi Kafka Connect image](https://strimzi.io/docs/master/#creating-new-image-from-base-str) with the Debezium connector jar file. 
+
+First download and extract the Debezium MySQL connector archive
+```shell
+curl https://repo1.maven.org/maven2/io/debezium/debezium-connector-mysql/0.10.0.Final/debezium-connector-mysql-0.10.0.Final-plugin.tar.gz \
+| tar xvz
+```
+
+Prepare a `Dockerfile` which adds those connector files to the Strimzi Kafka Connect image
+
+```shell
+cat <<EOF >Dockerfile
+FROM strimzi/kafka:latest-kafka-2.3.0
+USER root:root
+EXEC mkdir -p /opt/kafka/plugins/debezium
+COPY ./debezium-connector-mysql/ /opt/kafka/plugins/debezium/
+USER 1001
+EOF
+```
+
+Then build the image from that `Dockerfile` and push it to dockerhub.
+
+```shell
+# You can use your own dockerhub organization
+export DOCKER_ORG=tjbentley
+docker build . -t ${DOCKER_ORG}/connect-debezium
+docker push ${DOCKER_ORG}/connect-debezium
+```
+
+
+# Create the Connect cluster
+
+Now we can create a `KafkaConnect` cluster in Kubernetes:
+
+```shell
+cat <<EOF | kubectl apply -f -
+apiVersion: kafka.strimzi.io/v1beta1
+kind: KafkaConnect
+metadata:
+  name: my-connect-cluster
+  annotations:
+  # use-connector-resources configures this KafkaConnect
+  # to use KafkaConnector resources to avoid
+  # needing to call the Connect REST API directly
+    strimzi.io/use-connector-resources: "true"
+spec:
+  image: ${DOCKER_ORG}/connect-debezium
+  replicas: 1
+  bootstrapServers: my-cluster-kafka-bootstrap:9093
+  tls:
+    trustedCertificates:
+      - secretName: my-cluster-cluster-ca-cert
+        certificate: ca.crt
+EOF
+```
+
+It's worth pointing out a couple of things about the above resource:
+
+* The `strimzi.io/use-connector-resources: "true"` annotation tells the cluster operator that this `KafkaConnect` is managed via `KafkaConnector` resources.
```

Comment:
I don't get the point of this sentence. Anyway, the CO is going to deploy a Kafka Connect cluster via the `KafkaConnect` resource; then you can create connectors via `KafkaConnector`. What do you mean by "this `KafkaConnect` is managed via `KafkaConnector` resources" ... I missed the point of "is managed via".

Reply:
Would "will be managed using `KafkaConnector` resources." be clearer @ppatierno ?

Reply:
What sounds weird to me (which could actually be related to not being a native English) is that you want to say that `KafkaConnect` instance is managed via/using `KafkaConnector` resources. I think that from your point of view,  "managing" a Kafka Connect cluster means add/removing/configuring Kafka connectors so the "management" isn't via HTTP API but via connector resources. I see your point but it really sounds weird to me (again I don't know how it sounds in English). I would just highlight that connectors are created/deleted/configured in Kafka Connect via Kafka Connectors resources.

Reply:
So how about this: 
> The `strimzi.io/use-connector-resources: "true"` annotation tells the cluster operator that `KafkaConnector` resources will be used to configure connectors within this Kafka Connect cluster.

Reply:
That sounds great to me!